spec.emu

```diff
@@ -71,7 +71,7 @@ contributors: Google, Ecma International
       <emu-alg>
         1. If _type_ is `"language"`, then
           1. If _code_ does not matches the `unicode_language_id` production, throw a *RangeError* exceptipon.
-          1. Let _code_ be the result of mapping _code_ to lower case as described in <emu-xref href="#sec-case-sensitivity-and-case-mapping"></emu-xref>.
+          1. Set _code_ to CanonicalizeUnicodeLocaleId(_code_).
```

Comment:
`CanonicalizeUnicodeLocaleId` expects that its input is a valid string according to `IsStructurallyValidLanguageTag`, so the previous step should be replaced with `IsStructurallyValidLanguageTag`.

Reply:
Or rather we want to check if it matches `unicode_language_id` and `IsStructurallyValidLanguageTag` returns true, so locale ids with extension or private-use subtags aren't allowed.

Reply:
ok. Let me merge this PR AS IS first then we move on from there.